### PR TITLE
Move conversion deployment to direct-csi binary

### PR DESCRIPTION
**Pull request by Praveenrajmani**
- This removes conversion webhook as a separate deployment
- Client side upgrades are removed
- Conversion webhook is moved to node-driver and controller pods running in direct-csi-min-io namespace

### CODE REVIEW

File: pkg/converter/webhook.go [hunk 1]
```diff
@@ -63,11 +63,12 @@ func ServeConversionWebhook(ctx context.Context) error {
 		return lErr
 	}
 
-	klog.V(2).Infof("Starting conversion webhook server in port: %s", port)
-	if err := server.ServeTLS(listener, "", ""); err != nil {
-		klog.Errorf("Failed to listen and serve conversion webhook server: %v", err)
-		return err
-	}
+	go func() {
+		klog.V(2).Infof("Starting conversion webhook server in port: %s", port)
```
Comment (wlan0) [inline pkg/converter/webhook.go:67]: V(3)

**Review by wlan0 — APPROVED** [review]

File: pkg/node/discovery/discovery.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"k8s.io/klog/v2"
 	"path/filepath"
 	"strings"
+	// "time"
```
Comment (balamurugana) [inline pkg/node/discovery/discovery.go:25]: Remove this comment

File: pkg/node/discovery/discovery.go [hunk 3]
```diff
@@ -65,6 +66,7 @@ func NewDiscovery(ctx context.Context, identity, nodeID, rack, zone, region stri
 		driveTopology:   topologies,
 	}
 
+	// time.Sleep(15*time.Second)
```
Comment (balamurugana) [inline pkg/node/discovery/discovery.go:69]: Remove this comment

File: pkg/installer/uninstall.go [hunk 4]
```diff
@@ -166,16 +167,43 @@ func DeleteDeployment(ctx context.Context, identity, name string) error {
...
 	}
 	return nil
 }
+
+func DeleteConversionSecrets(ctx context.Context, identity string) error {
+	secretsClient := utils.GetKubeClient().CoreV1().Secrets(utils.SanitizeKubeResourceName(identity))
+	if err := secretsClient.Delete(ctx, conversionKeyPair, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if err := secretsClient.Delete(ctx, conversionCACert, metav1.DeleteOptions{}); err != nil {
```
Comment (balamurugana) [inline pkg/installer/uninstall.go:189]: Just return without checking.

File: pkg/installer/install.go [hunk 5]
```diff
@@ -1130,59 +993,39 @@ func GetConversionCABundle(ctx context.Context, identity string, dryRun bool) ([
...
-func isConversionDeploymentReady(ctx context.Context, identity string) bool {
-	deploymentsClient := utils.GetKubeClient().AppsV1().Deployments(utils.SanitizeKubeResourceName(identity))
-	deployment, getErr := deploymentsClient.Get(ctx, conversionWebhookName, metav1.GetOptions{})
-	if getErr != nil {
-		klog.V(5).Info(getErr)
-		return false
+	if err := CreateOrUpdateConversionKeyPairSecret(ctx, identity, publicCertBytes, privateKeyBytes, dryRun); err != nil {
+		return err
+	}
+	if err := CreateOrUpdateConversionCACertSecret(ctx, identity, caCertBytes, dryRun); err != nil {
```
Comment (balamurugana) [inline pkg/installer/install.go:1026]: Just return without checking

File: pkg/installer/install.go [hunk 6]
```diff
@@ -1130,59 +993,39 @@ func GetConversionCABundle(ctx context.Context, identity string, dryRun bool) ([
...
 }
 
-func GetConversionServiceName() string {
-	return conversionWebhookName
+func checkConversionSecrets(ctx context.Context, identity string) error {
+	secretsClient := utils.GetKubeClient().CoreV1().Secrets(utils.SanitizeKubeResourceName(identity))
+	if _, err := secretsClient.Get(ctx, conversionKeyPair, metav1.GetOptions{}); err != nil {
+		return err
+	}
+	if _, err := secretsClient.Get(ctx, conversionCACert, metav1.GetOptions{}); err != nil {
```
Comment (balamurugana) [inline pkg/installer/install.go:1001]: Just return without checking

File: cmd/direct-csi/cmd.go [hunk 7]
```diff
@@ -44,10 +44,9 @@ var (
 	controller           = false
 	driver               = false
 	procfs               = "/proc"
-	conversionWebhook    = false
-	conversionWebhookURL = ""
 	loopBackOnly         = false
 	showVersion          = false
+	conversionHealthzURL = ""
```
Comment (balamurugana) [inline cmd/direct-csi/cmd.go:49]: Why this funny name `healthz` ?

File: pkg/node/discovery/discovery.go [hunk 8]
```diff
@@ -22,7 +22,6 @@ import (
 	"k8s.io/klog/v2"
 	"path/filepath"
 	"strings"
-	// "time"
```
Comment (wlan0) [inline pkg/node/discovery/discovery.go]: Lets remove these comments

**Review by balamurugana — APPROVED** [review]

**Pull request merged**